ruby/ql/lib/codeql/ruby/typetracking/TypeTrackerSpecific.qll

```diff
@@ -190,3 +227,53 @@ predicate basicLoadStep(Node nodeFrom, Node nodeTo, string content) {
 class Boolean extends boolean {
   Boolean() { this = true or this = false }
 }
+
+private import SummaryComponentStack
+
+private predicate hasStoreSummary(
+  SummarizedCallable callable, TypeTrackerContent contents, SummaryComponent input,
+  SummaryComponent output
+) {
+  callable
+      .propagatesFlow(singleton(input),
+        push(SummaryComponent::content(contents), singleton(output)), true)
+}
+
+private predicate hasLoadSummary(
+  SummarizedCallable callable, TypeTrackerContent contents, SummaryComponent input,
+  SummaryComponent output
+) {
+  callable
+      .propagatesFlow(push(SummaryComponent::content(contents), singleton(input)),
+        singleton(output), true)
+}
+
+/**
+ * Gets a data flow node corresponding an argument or return value of `call`,
+ * as specified by `component`.
+ */
+bindingset[call, component]
+private DataFlowPublic::Node evaluateSummaryComponentLocal(
+  DataFlowPublic::CallNode call, SummaryComponent component
+) {
+  exists(DataFlowDispatch::ParameterPosition pos | component = SummaryComponent::argument(pos) |
+    exists(int i |
+      pos.isPositional(i) and
+      result = call.getPositionalArgument(i)
+    )
+    or
+    exists(string name |
+      pos.isKeyword(name) and
+      result = call.getKeywordArgument(name)
+    )
+    or
+    pos.isBlock() and
+    result = call.getBlock()
+    or
+    pos.isSelf() and
+    result = call.getReceiver()
```

Comment:
I think this can be replaced with
```ql
argumentPositionMatch(call, result, pos)
```

Reply:
Thanks! 👍  I was a little sad about having to do this here.